packages/browser/src/login/oidc/OidcLoginHandler.ts

```diff
@@ -44,6 +44,68 @@ import {
 
 import { IClient } from "@inrupt/oidc-client-ext";
 
+function isValidUrl(url: string): boolean {
+  try {
+    // Here, the URL constructor is just called to parse the given string and
+    // verify if it is a well-formed IRI.
+    // eslint-disable-next-line no-new
+    new URL(url);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function handleRegistration(
+  options: ILoginOptions,
+  issuerConfig: IIssuerConfig,
+  storageUtility: IStorageUtility,
+  clientRegistrar: IClientRegistrar
+): Promise<IClient> {
+  if (
+    options.clientId === undefined ||
+    (issuerConfig.solidOidcSupported === false && isValidUrl(options.clientId))
+  ) {
+    // If no client_id is provided, the client must go through DCR. Whether the
+    // Identity Provider supports Solid-OIDC or not will only change the value
+    // of the provided client_id.
+    // The tricky case is if a client WebID (here, a client_id looking like an IRI)
+    // is provided, but the Identity Provider does not support Solid-OIDC. In this
+    // case, we must discard the provided client_id, and still go through dynamic
+    // client registration, because the Identity Provider will not recognize the
+    // provided client_id.
```

Comment:
I *think* this wording might be clearer (in conjunction with the new wording in the comment below too) - but see what you think...
```suggestion
    // If no client_id is provided, the client must go through DCR. If a
    // client_id is provided and it looks like a URI, yet the Identity Provider does *not* support Solid-OIDC, then
    // we also perform DCR (and discard the provided client_id).
```

Reply:
0f4c45b5d22c2d0fa8f3d8ccc9055cce155fae14